pkg/splunk/enterprise/workerpool.go

```diff
@@ -127,29 +128,32 @@ func (w *AppInstallWorkerPool) Start() {
 
 					// add the job to the jobs channel
 					w.JobsChan <- app
-					jobsAdded++
+					jobsParsed++
+					actualJobsAdded = true
 				} else {
 					_ = fmt.Errorf("not enough space to accomodate app:%s, availableDiskSpace:%d", app.AppName, w.availableDiskSpace)
 					continue
 				}
 			case enterpriseApi.DownloadComplete:
 				// This is done in case if all the apps were already downloaded, we would need to
 				// break from outer loop and close the channel as there is no job to add.
-				jobsAdded++
+				jobsParsed++
 				continue
 			default:
 				continue
 			}
 		}
 
 		// all the jobs are added to the channel, we are done here
-		if jobsAdded == len(w.JobList) {
+		if jobsParsed == len(w.JobList) {
 			break
 		}
 	}
 
-	// close the jobs channel now
-	close(w.JobsChan)
+	// close the jobs channel now, only if it is not closed yet
+	if actualJobsAdded {
+		close(w.JobsChan)
```

Comment:
For my knowledge: if no jobs are added to the channel, it never gets opened?

Reply:
yup, right